tests/compile-fail/for_loop.rs

```diff
@@ -209,4 +209,20 @@ fn main() {
     let mut index = 0;
     for _v in &vec { index += 1 }
     println!("index: {}", index);
+
+    let m : HashMap<u64, u64> = HashMap::new();
+    for (_, v) in &m {
+        //~^ you seem to want to iterate on a map's values
+        //~| HELP use the corresponding method
+        //~| SUGGESTION for v in &m.values()
+        let _v = v;
+    }
+
+    let rm = &m;
+    for (k, _) in rm {
+        //~^ you seem to want to iterate on a map's keys
+        //~| HELP use the corresponding method
+        //~| SUGGESTION for k in rm.keys()
+        let _k = k;
+    }
 }
```

Comment:
I'd like to see a test case using a named wild card that is used anyway, e.g. `for (_k, v) in &map { let _x = _k; .. }`.